src/components/Form/Form.tsx

```diff
@@ -59,12 +63,32 @@ const Form = <T extends FormDetails>({
       }
     }
 
-    const errors = await props.onSubmit(state.form);
+    const errors = await props.onSubmit(state.form, buttonId);
     if (errors) {
       dispatch({
         type: "set_errors",
         errors: { ...state.errors, ...errors },
       });
+    } else {
+      setIsDirty(false);
+    }
```

Comment:
should this be part of the else block necessarily?

if `errors` are present, and even when submitted but for add more, in both cases we should set dirty back to false right? since when validation error happens, we expect the user to change something before re-submitting right?

Reply:
It doesn't matter in this component, as we are handling errors state validation in the submit function so it don't allow form to move forward to update isDirty state.
![image](https://github.com/user-attachments/assets/41049fc4-2d47-4d9b-83d1-859cf01b90c4)

Reply:
other than this I had made all the requested changes.